Libraries/LibGfx/Filter.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2025, Lucien Fiorini <lucienfiorini@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGfx/Filter.h>
+#include <LibGfx/FilterSkia.h>
+
+namespace Gfx {
+
+static FilterImplSkia impl_factory;
```

Comment:
as for now it seems it could be enough just to make `FilterSkia.cpp` implement `Filter.h` directly? (or is there some non-immediate gain in keeping this factory indirection?) 🤔

Reply:
The problem is that I need to define an accessor somewhere to retrieve the underlying skia filter, ideally in a separate skia specific header file like it's done for Gfx::Path to offload implementation specific accessors from the main header file. While I agree that it seems overkill to do it like this, I can't think of a better way that remains consistent with the rest of the codebase.

Reply:
for Path we do forward declaration of SkPath in PathSkia.h, can we do similar thing here for Filter?

Reply:
> for Path we do forward declaration of SkPath in PathSkia.h, can we do similar thing here for Filter?

Do you mean something like this ?

```c++
template<typename T>
class sk_sp<T>;

class SkImageFilter;

// Inside the Filter class in Filter.h:
NonnullOwnPtr<sk_sp<SkImageFilter>> m_impl;

```

Or is it fine to expose [SkRefCnt.h](https://api.skia.org/SkRefCnt_8h_source.html) to have direct access to the `sk_sp` shared pointer and avoid the need to have our unique pointer on top of a shared pointer.

Reply:
we can store sk_sp<SkImageFilter> in m_impl and define a getter `SkImageFilter& sk_image_filter() const;`. this way we won't have to use `sk_sp` in .h files. you can check PaintingSurface class that is implemented this way.

Reply:
The problem is that `paint.setImageFilter` needs the full sk_sp<SkImageFilter> down the line so exposing a reference won't work. Aside from declaring `sk_sp`, there is also the option to define FilterImpl in a header file that is only included privately in Skia specific `.cpp` files and put all the accessors in there and then provide a getter `FilterImpl *impl() const` in `Filter`.

Reply:
I pushed the changes :)